rpcs3/util/simd.hpp

```diff
@@ -1827,10 +1829,6 @@ inline v128 gv_fmafs(const v128& a, const v128& b, const v128& c)
 #elif defined(__FMA4__)
 	return _mm_macc_ps(a, b, c);
 #elif defined(ARCH_X64)
-	// This is inaccurate implementation
-#ifdef __AVX__
```

Comment:
This will have measurable performance effects on self-compiled builds with AVX support.

Reply:
I removed it because it doesn't work.
```
In file included from C:/src/rpcs3/rpcs3/Emu/Cell/SPURecompiler.cpp:28:
C:/src/rpcs3/rpcs3/util/simd.hpp: In function 'v128 gv_fmafs(const v128&, const v128&, const v128&)':
C:/src/rpcs3/rpcs3/util/simd.hpp:1834:85: error: cannot convert '__m256d' to '__m256'
 1834 |         const __m128 r = _mm256_cvtpd_ps(_mm256_add_ps(_mm256_mul_ps(_mm256_cvtps_pd(a), _mm256_cvtps_pd(b)), _mm256_cvtps_pd(c)));
      |                                                                      ~~~~~~~~~~~~~~~^~~
      |                                                                                     |
      |                                                                                     __m256d
In file included from C:/msys64/mingw64/lib/gcc/x86_64-w64-mingw32/13.1.0/include/immintrin.h:43,
                 from C:/src/rpcs3/rpcs3/util/tsc.hpp:8,
                 from C:/src/rpcs3/rpcs3/util/asm.hpp:4,
                 from C:/src/rpcs3/rpcs3/util/simd.hpp:6:
C:/msys64/mingw64/lib/gcc/x86_64-w64-mingw32/13.1.0/include/avxintrin.h:318:23: note:   initializing argument 1 of '__m256 _mm256_mul_ps(__m256, __m256)'
  318 | _mm256_mul_ps (__m256 __A, __m256 __B)
      |                ~~~~~~~^~~
```

Reply:
Ok, I'll leave the rest of the review to neko who is more familiar with the setup here.

Reply:
You can use std::bit_cast instesd _mm256_cvtps_pd.

Reply:
Sorry, I don't know how. What I tried didn't work.

Reply:
You can just leave it in. We shouldn't have AVX instructions enabled in CICD so it isn't a problem. Deleting it just hides it from view so nobody will ever fix whatever is wrong with it. You can also add compiler guards around it if needed.

Reply:
Reverted and enabled for Clang only, because it doesn't compile with GCC and MSVC.

Reply:
I pushed a fix for this issue. Please retest.

Reply:
Works now.